### DISCUSSION TITLE
Transmute element type into a new axis

### DISCUSSION THREAD

**Discussion by BTOdell** [rust-ndarray/ndarray#1383] [Q&A]
I have an `ArrayView<RGB8, Ix2>` where `RGB8` if from the `rgb` crate. This represents an RGB image with a width and height. I have my images in this format because it makes it easier to convert to a OpenCV Mat type for image processing. However, in some situations I need the data transmuted into a `ArrayView<u8, Ix3>` where the new axis is of length 3 for the red, green, blue color channels.

The array memory should already be in the correct format, so ideally I shouldn't have to iterate through every element and "expand" it manually into a third axis. Is there a way to do this?

**Comment by bluss** [comment]
This is a wanted feature. We have a proof of concept of this somewhere.
